tsdb/index/tsi1/index.go

```diff
@@ -277,33 +278,21 @@ func (i *Index) Open() error {
 		i.partitions[j] = p
 	}
 
+	defer i.cleanUpFail(&rErr)
```

Comment:
I think we want to defer this only after the `Wait` on errGroup - otherwise we could accidentally add a return and run `cleanUpFail` concurrently with the partition opening, which would be bad.

As I understand it, the partition type assumes `Close` is never called concurrently with `Open`.

Reply:
Fix pushed.